martin/src/pg/configurator.rs

```diff
@@ -232,12 +234,57 @@ impl PgBuilder {
     }
 }
 
+/// Try to find any ID column in a list of table columns (properties) that match one of the given `id_column` values.
+/// If found, modify `id_column` value on the table info.
+fn update_id_column(id: &str, inf: &mut TableInfo, auto_tables: &PgBuilderAuto) {
+    let Some(props) = inf.properties.as_mut() else { return };
+    let Some(try_columns) = &auto_tables.id_columns else { return };
+
+    for key in try_columns.iter() {
+        let (column, typ) = if let Some(typ) = props.get(key) {
+            (key, typ)
+        } else {
+            match find_kv_ignore_case(props, key) {
+                Ok(Some(result)) => {
+                    info!("For source {id}, id_column '{key}' was not found, but found '{result}' instead.");
+                    (result, props.get(result).unwrap())
+                }
+                Ok(None) => continue,
+                Err(multiple) => {
+                    error!("Unable to configure source {id} because id_column '{key}' has no exact match or more than one potential matches: {}", multiple.join(", "));
+                    continue;
+                }
+            }
+        };
+        // ID column can be any integer type as defined in
+        // https://github.com/postgis/postgis/blob/559c95d85564fb74fa9e3b7eafb74851810610da/postgis/mvt.c#L387C4-L387C66
+        if typ != "int4" && typ != "int8" && typ != "int2" {
+            warn!("Unable to use column `{key}` in table {}.{} as a tile feature ID because it has a non-integer type `{typ}`.", inf.schema, inf.table);
+            continue;
+        }
+
+        inf.id_column = Some(column.to_string());
+        let mut final_props = props.clone();
```

Comment:
@nyurik as you sai earlier that we should avoid clone whenever possible. can this clone be avoided?

Reply:
it is more of a "don't clone unless you have to" :)

I agree that here it would be better here without the clone, but the issue is that you have a mutable ref to a hashmap - `props`, plus you have a readonly ref to one of the keys in that hashmap - `column` - and Rust won't let you have both readonly and mutable refs to the same object at the same time.  If you can figure out a way around that, sure :)

Reply:
Yeah, I also searched a little and found that this was only way to do it. So, for now let's resolve it as it is.